### PR TITLE
Hotfix/issue209: Unit testing for RTP Package and decoding

### PR DESCRIPTION
**Pull request by KanishkT123**
## Checklist
***This checklist is used to make sure that common guidelines for a pull request are followed.***

- [X] **I have read the [contribution guidelines](../blob/master/CONTRIBUTING.md).**
- [X] Title of the pull request is clear and informative.
- [X] If applicable, the code is properly documented.
- [X] The code builds without any errors.
- [X] All unit and integration tests pass.

## Description
Added a unit test for the decoder which implicitly checks to see that the rtp packaging and rendering works as expected and explicitly checks to see that the decoding works as expected.

## Related Issues and PRs
#209 is resolved by this PR.

### CODE REVIEW

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp [hunk 1]
```diff
@@ -10,6 +10,9 @@
 #include "DeviceResources.h"
 #include "directx_buffer_capturer.h"
 #include "opengl_buffer_capturer.h"
+#include "webrtc\modules\rtp_rtcp\source\rtp_format_h264.h"
+#include "webrtc\modules\rtp_rtcp\source\rtp_packet_to_send.h"
+#include "webrtc\modules\rtp_rtcp\include\rtp_header_extension_map.h"
```
Comment (anderm) [inline Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: These are not needed

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp [hunk 2]
```diff
@@ -442,3 +445,81 @@ TEST(BufferCapturerTests, CaptureFrameStereoUsingDirectXBufferCapturer)
 			capturer->staging_frame_buffer_.Get(), 0);
 	}
 }
+
+// --------------------------------------------------------------
+// Decoder tests
+// --------------------------------------------------------------
+TEST(DecoderTests, CanDecodeCorrectly) {
```
Comment (anderm) [inline Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: Can you rename the unit test to RtpHeaderFramePredictionTest? That is the core of this test not the decode itself, we have plenty of decoder tests inside webrtc.

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp [hunk 3]
```diff
@@ -442,3 +445,81 @@ TEST(BufferCapturerTests, CaptureFrameStereoUsingDirectXBufferCapturer)
...
+	//Test implementation initializes decoder implicitly
```
Comment (anderm) [inline Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: nitpick - space after //

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp [hunk 4]
```diff
@@ -442,3 +445,81 @@ TEST(BufferCapturerTests, CaptureFrameStereoUsingDirectXBufferCapturer)
...
+		buffer->width(), buffer->height());
+
+	// Set RGB frame 
+	h264TestImpl->input_frame_.get()->set_frame_buffer(rgbBuffer);
+
+	// Encode frame
+	h264TestImpl->encoder_->Encode(*h264TestImpl->input_frame_, nullptr, nullptr);
+	EncodedImage encodedFrame;
+
+	//Set frame prediction timestamp to 1 (arbitrary value)
```
Comment (anderm) [inline Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: Same here, space after //

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp [hunk 5]
```diff
@@ -442,3 +445,81 @@ TEST(BufferCapturerTests, CaptureFrameStereoUsingDirectXBufferCapturer)
...
+	encodedFrame.prediction_timestamp_ = 1;
+
+	// Extract encoded_frame from the encoder
+	h264TestImpl->WaitForEncodedFrame(&encodedFrame);
+
+	// Check if we have a complete frame with lengh > 0
+	ASSERT_TRUE(encodedFrame._completeFrame);
+	ASSERT_TRUE(encodedFrame._length > 0);
+
+	//The first frame needs to be a keyframe for decoding
```
Comment (anderm) [inline Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: A few more from this line until 515. Add a space after //

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp (hunk 1 above)
Comment (KanishkT123) [reply Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: Removed them!

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp (hunk 2 above)
Comment (KanishkT123) [reply Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: Done!

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp (hunk 5 above)
Comment (KanishkT123) [reply Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: All done!

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp [hunk 6]
```diff
@@ -442,3 +442,95 @@ TEST(BufferCapturerTests, CaptureFrameStereoUsingDirectXBufferCapturer)
 			capturer->staging_frame_buffer_.Get(), 0);
 	}
 }
+
+// --------------------------------------------------------------
+// Decoder tests
+// --------------------------------------------------------------
+TEST(RtpHeaderFramePredictionTest, CanPredictCorrectly) {
```
Comment (anderm) [inline Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: CanPredictCorrectly might be a bit misleading, let's rename to SendReceiveFramePredictionTimestamps

File: Samples/Server/NativeServer.Tests/NativeServersTests.cpp [hunk 7]
```diff
@@ -442,3 +442,95 @@ TEST(BufferCapturerTests, CaptureFrameStereoUsingDirectXBufferCapturer)
...
+	// Test implementation initializes decoder implicitly
+	auto h264TestImpl = new H264TestImpl();
+	h264TestImpl->SetEncoderHWEnabled(true);
+	int defaultCodecWidth = 1280;
+	int defaultCodecHeight= 720;
+
+	//Generate a frame using a framegenerator class
```
Comment (anderm) [inline Samples/Server/NativeServer.Tests/NativeServersTests.cpp]: Space after //

**Pull request merged**